Mandatory pages css

annasvrv opened the pull request.

annasvrv committed:
Created css file and added styles for homepage

annasvrv committed:
added linked css files and normalized css; added some classes

annasvrv committed:
add new images

annasvrv committed:
minor class fixes

annasvrv committed:
changed cta classes and linked css files

annasvrv committed:
added styles for catalog page

annasvrv committed:
Linked css files and added some classes

annasvrv committed:
Added styles for catalog-item.html page

annasvrv committed:
fixed margin bottom in nftcard btns

annasvrv committed:
removed border from nft-cat-gtbtn

annasvrv committed:
Added some classes and linked css files

annasvrv committed:
added styles for contact us page

annasvrv committed:
changed text area size in the form

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+  text-shadow: 0px 0px 1px var(--btn-primary-color);
+}
+
+/* I couldn't add another angle border effect */
```
we will talk about css magic later

annasvrv replied:
I will leave it as it is till next css lesson about magic ))

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+* {
+  padding: 0;
+  margin: 0;
+  background-color: rgba(4, 11, 17, 1);
```
Translate global text and background styles from the universal selector * to the body selector:

Why: Using the universal selector * can negatively impact performance and can inadvertently apply styles to all elements, including those you may not intend to style. Applying general styles directly to the body is more efficient and ensures that only the base of your document is affected.

[Read more about CSS performance and the universal selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Universal_selectors).

annasvrv replied:
Thank you for this comment. Now I see how * can impact whole code. 

#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+  cursor: pointer;
+}
+
+span {
```
Better not to style tags like <span> or other elements directly:

Applying styles directly to tags like span can lead to unintended consequences, especially if those tags are used in various contexts throughout the site. It's better to use classes or IDs for more specific targeting, which helps maintain consistent and predictable styling.

Example:

```
.highlighted-text {
    color: red;
}
```

This way, you can control which specific span elements are affected by your styles.

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+}
+
+.hero-title,
+.hero-title span {
```
.hero-title span does not need the same styles as .hero-title since text styles are inherited from the parent:

CSS inheritance automatically passes certain properties (like color, font-family, etc.) from parent elements to their children, so there’s no need to reapply the same styles unless you intend to override them.

[Learn more about CSS inheritance](https://developer.mozilla.org/en-US/docs/Web/CSS/inheritance).

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+  background-color: var(--secondary-color);
+}
+
+.hero-box * {
```
It’s better not to use the universal selector:

The universal selector (*) can have a significant performance cost and may apply styles more broadly than intended. It's more efficient and safer to target specific elements or classes.

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+}
+
+.arrow-greater-btn {
+  float: inline-end;
```
float is not the most appropriate property for positioning here:

The float property was originally designed for text wrapping around images, not for complex layouts. In modern web design, it's better to use Flexbox or Grid for layout and positioning, as they offer more control and are more predictable.

annasvrv replied:
I tried to fixed it. Not sure if it's a correct way. 

#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+/* Change position of the marker and image */
+
+.workflow-card::marker {
+  content: "STEP " counter(list-item);
```
Excellent use of counter for the list. :+1:

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+
+/* Call Action section - coloful */
+
+.CTA-section {
```
Avoid using uppercase letters in class names:
Class names in HTML and CSS are case-sensitive, and using lowercase letters is a standard convention that helps avoid potential errors and inconsistencies.

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+
+.catalog-footer {
+  background-color: var(--secondary-color);
+  padding-top: -40px;
```
Padding cannot be negative:

In CSS, padding represents the space between the content and the border of an element and cannot be negative. Negative padding would not make sense, as it would imply that the content overlaps the border, which is not possible.

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1091 @@
+  padding: 0;
+  margin: 0;
+  background-color: rgba(4, 11, 17, 1);
+  font-family: "Bakbak One", sans-serif;
```
Provide fallback fonts and family types at the end of the font list. 
For example: `font-family: "Roboto", Arial, sans-serif;`

annasvrv replied:
#fixed

broacademy commented on img/btn-icons/Minus.svg:
Use lowercase to files naming.

annasvrv replied:
#fixed

broacademy commented on img/btn-icons/tw-footer.png:
Try to use svg format for this icon.

annasvrv replied:
#fixed

broacademy commented:
Homework checklist: 

- [x] ❤️ 2.1. Single CSS File.
- [x] 💛 2.2. Include Normalize.css
- [x] ❤️ 2.3. All fonts used in the design are connected to the pages.
- [ ] ❤️ 2.4. Provide fallback fonts and family types at the end of the font list. 
- [x] ❤️ 2.5. Do not use !important in CSS.
- [x] ❤️ 2.6. Do not use #id for styling.
- [x] 💛 2.7. Avoid nesting selectors more than two levels deep.
- [ ] 💛 2.8. Avoid styling tags directly
- [x] ❤️ 2.10. Use consistent units for element sizes and positioning.
- [x] 💛 2.11. Colors should be in a consistent format (hex or rgba).

Overall, great work! Here’s what I liked:

- Your foundational styles are very well done
- Excellent use of counters for lists. 👍
- The files are correctly linked.
- Normalize.css is properly included.
- Fonts are correctly linked.

What needs to be improved overall:

- It’s better not to use the universal selector (*)
- Better not to style tags like or other elements directly
- Provide fallback fonts and family types at the end of the font list.

annasvrv committed:
fixed btn classes, changed roadmap position phase list

annasvrv committed:
fixed styles

annasvrv committed:
changed buttons classes, fixed capitalized class name

annasvrv committed:
fixed buttons styles

annasvrv committed:
added favicon, fixed classes and buttons

annasvrv committed:
fixed buttons styles

annasvrv committed:
updated images formats

annasvrv committed:
created and added favicons

annasvrv committed:
updated img folder

annasvrv committed:
Linked favicons, styles, fonts in the page.

Changes are made for additional pages.

annasvrv committed:
Updated header and footer in the page

annasvrv committed:
Updated cta section in the page

annasvrv committed:
Updated breadcrumbs section in the page.

Changes are mage for additional pages.

annasvrv committed:
Styled faq page.

Added div.q-box elem and mb-mt atributes. Updated details.

annasvrv committed:
Styled about page.

About cards need to be done.

annasvrv committed:
Updated marker url of faq section in css.

Changed img files to lower case and added a correct url path in css marker for details faq.

annasvrv committed:
Changed files name to lower case

annasvrv committed:
remove file Minus.svg

annasvrv committed:
remove file Plus.svg

broacademy commented on css/style.css:
```diff
@@ -1 +1,1087 @@
+  --main-background-color: rgba(13, 13, 17, 1);
+  --text-color: rgba(194, 195, 197, 1);
+  --inactive-color: rgba(136, 139, 142, 1);
+  --text-font: "Poppins", sans-serif;
```
Provide fallback fonts and family types at the end of the font list in custom properties too.  
For example: font-family: "Roboto", Arial, sans-serif;

annasvrv replied:
#fixed

broacademy commented on css/style.css:
```diff
@@ -1 +1,1087 @@
+
+body {
+  background-color: rgba(4, 11, 17, 1);
+  font-family: "Bakbak One", sans-serif, "Roboto", Arial, sans-serif;
```
Use here `--btn-header-font` variable. 
`"Roboto", Arial, sans-serif` is just example. In your case it is: 
font-family: "Bakbak One", Arial, sans-serif;

https://crystallize.com/blog/web-safe-fonts#:~:text=System%20Fonts%3A%20System%20fonts%20are,consistent%20rendering%20on%20different%20devices.

annasvrv replied:
#fixed

broacademy commented:
You're doing great, but please pay more attention to how the font-family property works and what web-safe fonts are. You can merge this pull request and correct the fonts in the next one when you submit your image assignment for review.
